catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java

```diff
@@ -78,14 +93,19 @@ protected SolrCacheMBean getCacheProxy()
 
     protected CatalogFacade getCatalog() throws InterruptedException {
         if (isProvider) {
-            return new Provider(getService(CatalogProvider.class));
+            return new Provider(catalogProvider);
         } else {
             // otherwise use default
-            return new Framework(getService(CatalogFramework.class));
+            return new Framework(catalogFramework);
         }
     }
 
-    protected FilterBuilder getFilterBuilder() throws InterruptedException {
-        return getService(FilterBuilder.class);
+    protected <T> T getServiceByFilter(Class<T> clazz, String filter)
```

Comment:
I like returning an Optional in cases where the code could otherwise return a null value.